lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb

```diff
@@ -72,10 +72,16 @@ def upload_binary(app_id, binary_path)
         UI.crash!("#{ErrorMessage::APK_NOT_FOUND}: #{binary_path}")
       end
 
-      # Uploads the binary
+      # Uploads the binary file if it has not already been uploaded
+      # Takes at least POLLING_INTERVAL_SECONDS between retrying uploads
       #
-      # Returns the release_id on a successful release.
-      # Returns nil if unable to upload.
+      # args
+      #   app_id - Firebase App ID
+      #   binary_path - Absolute path to you app's apk/ipa file
+      #
+      # Returns the release_id on a successful release, otherwise returns nil.
+      #
+      # Throws an error if the number of polling attempts exceeds MAX_POLLING_RETRIES
```

Comment:
I love this method description format you used: it really works for a method like this that could use some explanation. Did you come up with this format yourself or adapt it from somewhere?

Reply:
Not really, I couldn't find a general format that most people use for ruby unless they're using a docwriter so I just went with this. I'm planning on writing all the other method descriptions like this

Reply:
Sounds good - and feel free to pick and choose which pieces you include, for example leaving out the args part if it's self explanatory